### PR TITLE
Task/IATIIMPORT-227/optimize iati processor

### PR DESCRIPTION
**Pull request by jdeanquin-dg**

### CODE REVIEW

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java [hunk 1]
```diff
@@ -405,4 +643,407 @@ private String getPropertieFieldName(String lang,String langPackType) {
...
+        processForEachFilteredByType(locationConsumer, FieldType.LOCATION);
+    }
+
+    protected void processDateElementType(InternalDocument document, Element element) throws ParseException {
+        NodeList nodes = element.getElementsByTagName("activity-date");
+        for (int j = 0; j < nodes.getLength(); ++j) {
+            Element e = (Element) nodes.item(j);
+            Field field = getField(e, FieldType.DATE);
+            String localDate = e.getAttribute("iso-date");
+                String format = "yyyy-MM-dd";
```
Comment (gmutuhu) [inline import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: Please change to use the constant ISO_DATE defined in the same class.

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java [hunk 2]
```diff
@@ -67,22 +63,30 @@
...
-    protected final static Map<String,Map<String,Map<String,Map<String,String>> >>langPack = new HashMap();
+    protected final static Map<String, Map<String, Map<String, Map<String, String>>>> langPack = new HashMap();
+
+    private static final String ISO_DATE = "yyyy-MM-dd";
+    private SimpleDateFormat df = new SimpleDateFormat(ISO_DATE);
+
 
     @Override
-    public  void setFromDataStore(boolean fromDatastore) {
+    public void setFromDataStore(boolean fromDatastore) {
```
Comment (gmutuhu) [inline import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: There are two setters for fromDatastore: setFromDataStore and setFromDatastore. Please remove one.

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java [hunk 3]
```diff
@@ -293,27 +528,29 @@ protected Document getDocument(String fileName) throws ParserConfigurationExcept
         }
     }
 
-    public NodeList getNodeListFromXpath(Document xDoc, String xPathExpresion)  {
+    public NodeList getNodeListFromXpath(Document xDoc, String xPathExpresion) {
 
         XPath xPath = XPathFactory.newInstance().newXPath();
         try {
-            Node node= (Node)xPath.compile(xPathExpresion).evaluate(xDoc,XPathConstants.NODE);
+            Node node = (Node) xPath.compile(xPathExpresion).evaluate(xDoc, XPathConstants.NODE);
```
Comment (gmutuhu) [inline import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: The variable node is not used. Please remove

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java [hunk 4]
```diff
@@ -405,4 +643,407 @@ private String getPropertieFieldName(String lang,String langPackType) {
...
+            Element iatiActivity = (Element) nodeList.item(i);
+            //we set the grouping criteria
+            document.setGrouping(getStringFromElement(iatiActivity, DEFAULT_GROUPING_FIELD, "ref"));
+
+            String currency = !("".equals(iatiActivity.getAttribute("default-currency"))) ? iatiActivity.getAttribute
+                    ("default-currency") : this.getDefaultCurrency();
+            document.addStringField("default-currency", currency);
+            String defaultLanguageCode = !("".equals(iatiActivity.getAttribute("xml:lang"))) ? iatiActivity.getAttribute
+                    ("xml:lang") : this.getDefaultLanguage();
+            NodeList fieldNodeList;
```
Comment (gmutuhu) [inline import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: The variable fieldNodeList is not used.

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java [hunk 5]
```diff
@@ -248,13 +482,14 @@ public Boolean isValidInput() {
...
+
+    public String getExtraQueryVersion() {
+        return "@*[name()='iati-extra:version']='" + PROCESSOR_VERSION + "']";
     }
 
     protected Document getDocument(String fileName) throws ParserConfigurationException, SAXException, IOException {
         InputStream is = this.getClass().getResourceAsStream(fileName);
-        if(is == null){
+        if (is == null) {
             System.out.println("this field name is null:" + fileName);
```
Comment (gmutuhu) [inline import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: Use the logger for this

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java [hunk 6]
```diff
@@ -405,4 +643,407 @@ private String getPropertieFieldName(String lang,String langPackType) {
...
+    }
+
+
+    protected void processTransactionElementType(InternalDocument document, Element iatiActivity,
+                                                 Element activityProviderNode)  {
+        NodeList nodes;
+        nodes = iatiActivity.getElementsByTagName("transaction");
+        for (int j = 0; j < nodes.getLength(); ++j) {
+            String reference ;
+            Element e = (Element) nodes.item(j);
```
Comment (gmutuhu) [inline import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: Rename e to something more descriptive e.g transactionElement

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java (hunk 1 above)
Comment (jdeanquin-dg) [reply import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: Done

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java (hunk 2 above)
Comment (jdeanquin-dg) [reply import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: done

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java (hunk 3 above)
Comment (jdeanquin-dg) [reply import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: done

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java (hunk 4 above)
Comment (jdeanquin-dg) [reply import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: done

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java (hunk 5 above)
Comment (jdeanquin-dg) [reply import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: done

File: import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java (hunk 6 above)
Comment (jdeanquin-dg) [reply import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java]: done

**Pull request merged**